src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -310,59 +305,55 @@ int BATT_SMBUS::dataflash_write(uint16_t address, void *data, const unsigned len
 	memcpy(&tx_buf[2], data, length);
 
 	// code (1), byte_count (1), addr(2), data(32) + pec
-	int result = _interface->block_write(code, tx_buf, length + 2, false);
+	int ret = _interface->block_write(code, tx_buf, length + 2, false);
 
-	return result;
+	return ret;
 }
 
 int BATT_SMBUS::get_startup_info()
 {
-	int result = 0;
-
-	// The name field is 21 characters, add one for null terminator.
-	const unsigned name_length = 22;
+	int ret = PX4_OK;
 
 	// Read battery threshold params on startup.
-	param_get(param_find("BAT_CRIT_THR"), &_crit_thr);
-	param_get(param_find("BAT_LOW_THR"), &_low_thr);
-	param_get(param_find("BAT_EMERGEN_THR"), &_emergency_thr);
-	param_get(param_find("BAT_C_MULT"), &_c_mult);
-
-	// Try and get battery SBS info.
-	if (_manufacturer_name == nullptr) {
-		char man_name[name_length] = {};
-		result = manufacturer_name((uint8_t *)man_name, sizeof(man_name));
-
-		if (result != PX4_OK) {
-			PX4_DEBUG("Failed to get manufacturer name");
-			return PX4_ERROR;
-		}
+	param_get((param_t)(px4::params::BAT_CRIT_THR), &_crit_thr);
+	param_get((param_t)(px4::params::BAT_LOW_THR), &_low_thr);
+	param_get((param_t)(px4::params::BAT_EMERGEN_THR), &_emergency_thr);
+	param_get((param_t)(px4::params::BAT_C_MULT), &_c_mult);
 
-		_manufacturer_name = new char[sizeof(man_name)];
-	}
+	int32_t cell_count_param = 0;
+	param_get((param_t)(px4::params::BAT_N_CELLS), &cell_count_param);
+	_cell_count = (uint8_t)cell_count_param;
+
+	ret |= _interface->block_read(BATT_SMBUS_MANUFACTURER_NAME, _manufacturer_name, BATT_SMBUS_MANUFACTURER_NAME_SIZE,
+				      true);
+	_manufacturer_name[sizeof(_manufacturer_name) - 1] = '\0';
 
 	uint16_t serial_num;
-	result = _interface->read_word(BATT_SMBUS_SERIAL_NUMBER, serial_num);
+	ret |= _interface->read_word(BATT_SMBUS_SERIAL_NUMBER, serial_num);
 
 	uint16_t remaining_cap;
-	result |= _interface->read_word(BATT_SMBUS_REMAINING_CAPACITY, remaining_cap);
+	ret |= _interface->read_word(BATT_SMBUS_REMAINING_CAPACITY, remaining_cap);
 
 	uint16_t cycle_count;
-	result |= _interface->read_word(BATT_SMBUS_CYCLE_COUNT, cycle_count);
+	ret |= _interface->read_word(BATT_SMBUS_CYCLE_COUNT, cycle_count);
 
 	uint16_t full_cap;
-	result |= _interface->read_word(BATT_SMBUS_FULL_CHARGE_CAPACITY, full_cap);
+	ret |= _interface->read_word(BATT_SMBUS_FULL_CHARGE_CAPACITY, full_cap);
 
-	if (!result) {
+	uint16_t manufacture_date;
+	ret |= _interface->read_word(BATT_SMBUS_MANUFACTURE_DATE, manufacture_date);
+
+	if (!ret) {
 		_serial_number = serial_num;
 		_batt_startup_capacity = (uint16_t)((float)remaining_cap * _c_mult);
 		_cycle_count = cycle_count;
 		_batt_capacity = (uint16_t)((float)full_cap * _c_mult);
+		_manufacture_date = manufacture_date;
 	}
 
 	if (lifetime_data_flush() == PX4_OK) {
 		// Flush needs time to complete, otherwise device is busy. 100ms not enough, 200ms works.
-		px4_usleep(200000);
+		px4_usleep(200_ms);
```

Comment:
Be careful here if this is executed from the WQ thread (I2C bus). This is long enough to cause magnetometer or barometer timeouts if sharing the same bus.

Reply:
that actually is simple refectory, so if there was problem before, it will be still a problem.
I am not sure how to handle that comment....

Reply:
I'll take a closer look when I have some appropriate hardware on hand.